### DISCUSSION TITLE
Connecting to Databricks via Simba Odbc Driver

### DISCUSSION THREAD

**Discussion by blk99cobra18103** [nhibernate/nhibernate-core#3551] [Q&A]
I’m utilizing nHibernate/Unit of Work/Repository pattern to connect to a sql server database. The solution is built in a .net 6 c# blazor app. Everything works great.

There is a company directive to move to databricks so I’ve been trying to integrate nHibernate with the Simba ODBC driver. I’m able to pull data from my databricks tables. The issue arises when I try to insert/update records. When I call my unitOfWork.Commit() method, the data is added/updated in databricks. The problem arises when I continue to step through the code. I’m receiving the following errors right after the commit call.

Inserting Error: Nhibernate.StaleStateException ‘Unexpected row count -1; expected 1’.

```
            try
            {
                unitOfWork.BeginTransaction(dbConnectionString);
                int newAgencyId = await GetLastDimAgencyIdAsync();
                dimAgency.Id = newAgencyId + 1;
                var result = await dbContext.DimAgencyRepo.AddAsync(dimAgency);
                //await dbContext.DimAgencyRepo.FlushAsync();
                await unitOfWork.Commit();
                return (int)result;
            }
```

Updating Error: Nhibernate.StaleObjectStateException: 'Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect)

```
            try
            {
                unitOfWork.BeginTransaction(dbConnectionString);
                dbContext.DimAgencyRepo.Update(dimAgency);
                unitOfWork.CommitTransaction();
                return true;
            }
```

Note there is only one transaction happening, there are no other transactions involved. I’ve also tried to implement ‘versioning’ in my table/model/mapping class to no avail.

Any ideas on where to begin to trouble shoot would be greatly appreciated?

**Comment by fredericDelaporte** [comment]
It seems the database or accessing that database through ODBC does not yield back the affected row count, which causes NHibernate to consider the operation was unsuccessful.

**Reply by blk99cobra18103** [reply]
That was my original thought that databricks doesn't return the affected row count like sql server does.  I did research if there was some kind of system wide setting in databricks to set rowcounton but couldn't find anything.
